src/aind_nwb_utils/nwb_io.py

```diff
@@ -0,0 +1,50 @@
+"""io tools for nwb files"""
+
+import atexit
+import os
+import tempfile
+from pathlib import Path
+from typing import Union
+
+from hdmf_zarr import NWBZarrIO
+from pynwb import NWBHDF5IO
+
+
+def create_temp_nwb(
+    save_strategy=Union[NWBHDF5IO, NWBZarrIO], dir=None
+) -> Path:
+    """Create a temporary file and return the path
+
+    Parameters
+    ----------
+    save_strategy : Union[NWBHDF5IO, NWBZarrIO]
+        to determine if a temp file or directory should be created
+    dir : str
+        option to specify root directory
+    Returns
+    -------
+    str
+        the path to the temporary file
+    """
+    if save_strategy is NWBZarrIO:
+        temp_path = tempfile.mkdtemp(dir=dir)
+    else:
+        temp = tempfile.NamedTemporaryFile(delete=False, suffix=".nwb")
+        temp_path = temp.name
+        temp.close()
+
+    atexit.register(os.unlink, temp_path)
+    return Path(temp_path)
+
+
+def determine_io(nwb_path: Path) -> Union[NWBHDF5IO, NWBZarrIO]:
```

Comment:
surprised there is no existing NWB utility for this

Reply:
I did not see anything